egs/wsj/s5/utils/lang/make_subword_lexicon_fst.py

```diff
@@ -0,0 +1,159 @@
+#!/usr/bin/env python3
+
+# 2019 Dongji Gao
+
+from make_lexicon_fst import read_lexiconp
+import argparse
+import math
+
+
+def get_args():
+    parser = argparse.ArgumentParser(description="""This script creates a 
+        position depent subword lexicon FST from a origin lexicon""")
+    parser.add_argument('--sil-phone', type=str, help="""Text form of
+        optional-silence phone.""")
+    parser.add_argument('--sil-prob', type=float, default=0.0, help="""Probability
+        of silence between words (including the beginning and end of word sequence).
+        Must be in range [0.0, 1.0]""")
+    parser.add_argument('--sil-disambig', type=str, help="""Disambiguation symbol
+        to disambiguate silence, e.g. #5.""")
+    parser.add_argument('--position-dependent', action="store_true", help="""Whether the input lexicon
+        is position-dependent.""")
+    parser.add_argument("--separator", type=str, default="@@", help="""Separator indicates the
+        position of a subword in word.""")
+    parser.add_argument('lexiconp', type=str, help="""Filename of lexicon with
+        pronunciation probabilities (normally lexiconp.txt), with lines of the
+        form 'word prob p1 p2...', e.g. 'a, 1.0 ay'""")
+    args = parser.parse_args()
+    return args
+
+def write_fst_no_silence(lexicon):
+    pass
+
+def contain_disambig_symbol(phones):
+    return True if phones[-1][0] == "#" else False
+
+
+def write_fst_with_silence(lexicon, sil_phone, sil_prob, sil_disambig, position_dependent, separator):
```

Comment:
it'd be a good idea to have more documentation regarding what this function does. and maybe some of the sub-functions also.